lib/mongodb.js

```diff
@@ -401,6 +403,12 @@ MongoDB.prototype.fromDatabase = function(modelName, data) {
       if (data[p] instanceof mongodb.Binary) {
         // Convert the Binary into String
         data[p] = data[p].toString();
+      // SPIKE: properties that should be converted to string from ObjectID
+      // SPIKE: do the same for toDatabase
+      } else if (p === 'id' || prop.mongodb && prop.mongodb.dataType.match(/objectid/i)) {
```

Comment:
Please keep in mind that primary keys are not required to be called `id`. If you really need to identify primary keys here (which I think you should not need to), then you need to check if `prop.id` is set (I think). I also vaguely remember that `prop.id` can be a number to describe multi-column index (e.g. on GitHub, `userId` would have `id: 1` and `repoId` would have `id: 2` to define `(userId, repoId)` as the primary key). I am not sure if those numeric values are indexed from `0` or `1` though.